OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -66,37 +65,40 @@ IEnumerable<IRenderable> IRenderAboveShroudWhenSelected.RenderAboveShroud(Actor
 		{
 			var force = Game.GetModifierKeys().HasModifier(Modifiers.Alt);
 			if ((lifetime <= 0 || --lifetime <= 0) && !force)
-				yield break;
+				return new IRenderable[0];
 
 			if (!(force || Game.Settings.Game.DrawTargetLine))
-				yield break;
+				return new IRenderable[0];
 
-			if (targets == null || targets.Count == 0)
-				yield break;
+			if (self.GetCurrentActivity() == null)
+				return new IRenderable[0];
 
-			foreach (var target in targets)
-			{
-				if (target.Type == TargetType.Invalid)
-					continue;
+			if (self.GetCurrentActivity().GetTargets(self).Count() <= 0)
+				return new IRenderable[0];
+
+			var validTargets = new List<WPos>();
+			validTargets.Add(self.CenterPosition);
 
-				yield return new TargetLineRenderable(new[] { self.CenterPosition, target.CenterPosition }, c);
+			foreach (var target in self.GetCurrentActivity().GetTargets(self))
+			{
+				validTargets.Add(target.CenterPosition);
 			}
+
+			return new[] { (IRenderable)new TargetLineRenderable(validTargets, c) };
 		}
 
 		void INotifyBecomingIdle.OnBecomingIdle(Actor a)
 		{
-			if (a.IsIdle)
-				targets = null;
```

Comment:
Please remove this method and the interface it is implementing (INotifyBecomingIdle) since the class isn't using it anymore.